src/libraries/System.Private.Uri/src/System/DomainNameHelper.cs

```diff
@@ -373,11 +323,10 @@ private static unsafe bool IsIdnAce(char* input, int index)
                         // check ace validity
                         try
                         {
-                            unicodeEqvlHost += s_idnMapping.GetUnicode(unescapedHostname, curPos, newPos - curPos);
+                            dest.AppendString(s_idnMapping.GetUnicode(hostname, curPos, newPos - curPos));
```

Comment:
What is "AppendString"?  Why isn't this just using ValueStringBuilder.Append?

Reply:
https://github.com/dotnet/runtime/pull/34864#discussion_r412147723. I removed the VSBExtensions file as you suggested as it's not needed for this PR.